debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordState.java

```diff
@@ -291,7 +296,12 @@ private R addFields(List<FieldReference> additionalFields, R originalRecord, R u
         }
 
         for (FieldReference fieldReference : additionalFields) {
-            updatedValue = updateValue(fieldReference, updatedValue, originalRecordValue);
+            try {
+                updatedValue = updateValue(fieldReference, updatedValue, originalRecordValue);
+            }
+            catch (DataException e) {
+                LOGGER.debug("Field {} not found in the record {}. Skipping it", fieldReference.field, originalRecordValue);
```

Comment:
Please check `io.debezium.transforms.ExtractNewRecordState.FieldReference.isInSchema(Schema)`.
Could the same approach be used instead of relying on exception handling?